pkg/flow/internal/controller/loader.go

```diff
@@ -56,6 +58,19 @@ func NewLoader(globals ComponentGlobals) *Loader {
 	if globals.Registerer != nil {
 		globals.Registerer.MustRegister(cc)
 	}
+
+	globals.Clusterer.Node.Observe(ckit.FuncObserver(func(peers []peer.Peer) (reregister bool) {
+		for _, cmp := range l.Components() {
```

Comment:
You have a memory leak in your design: if a component registered for cluster updates is removed from the config file, its registration will persist forever. 

I think the marker interface you had before is probably preferable over the explicit registration design because it makes it harder for memory to leak.